backend/api/endpoints/auth.py

```
@@ -0,0 +1,53 @@
+class LoginRequest(BaseModel):
+    username: str
+    password: str
+
+class User(BaseModel):
+    id: str
+    username: str
+    role: str
+    email: Optional[str] = None
+
+@router.post("/login")
+async def login(credentials: LoginRequest, db: Session = Depends(get_db)):
```

Ruff (0.14.5)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)